sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java

```diff
@@ -638,4 +650,20 @@ private void throwIfCopyOperationStatusInvalid(CopyOperationResult copyResult) {
         }
     }
 
+     /**
+      *  Helper method that throws a {@link HttpResponseException} if {@link ModelInfo#getStatus()} is
+      *  {@link com.azure.ai.formrecognizer.implementation.models.ModelStatus#INVALID}.
+      *
+      * @param customModel The response returned from the service.
+      */
+    private void throwIfModelStatusInvalid(Model customModel) {
+        if (ModelStatus.INVALID.equals(customModel.getModelInfo().getStatus())) {
+            List<ErrorInformation> errorInformationList = customModel.getTrainResult().getErrors();
+            if (!CoreUtils.isNullOrEmpty(errorInformationList)) {
+                throw logger.logExceptionAsError(new HttpResponseException(
```

Comment:
Should this be a `HttpResponseException` or just `AzureException`? `HttpResponseException` is usually an error returned by the HTTP call with a status code not equal to 2xx.

From [HttpResponseException JavaDoc](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/apidocs/com/azure/core/exception/HttpResponseException.html):
>The exception thrown when an unsuccessful response is received with http status code (e.g. 3XX, 4XX, 5XX) from the service request.

Reply:
For failures for service requests (3XX, 4XX etc) we have [ErroeResponseException](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/FormRecognizerClientImpl.java#L143).

Using the equivalent for other languages [`RequestFailedException`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/src/CopyModelOperation.cs#L163) and [`HttpResponseError`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py#L111)

Reply:
Discussed offline.
Maintaining the Javadoc for `HttpResponseException` and TA convention for exception handling, consider adding a new Exception Type or use AzureException.
https://github.com/Azure/azure-sdk-for-java/issues/11705

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/11720